mlir/lib/Transforms/Utils/RegionUtils.cpp

```diff
@@ -674,6 +679,64 @@ static bool ableToUpdatePredOperands(Block *block) {
   return true;
 }
 
+/// Prunes the redundant list of arguments. E.g., if we are passing an argument
+/// list like [x, y, z, x] this would return [x, y, z] and it would update the
+/// `block` (to whom the argument are passed to) accordingly
+static void
+pruneRedundantArguments(SmallVector<SmallVector<Value, 8>, 2> &newArguments,
```

Comment:
Either take `SmallVectorImpl` or return the vector

Reply:
I ended up returning the `SmallVector<SmallVector<Value, 8>, 2>`. I tried to use the `SmallVectorImpl`, but the problem is that I am dealing with a vector-of-vectors, so it was a bit more complicated.